QCflow/run_opt_neutral.py

```diff
@@ -68,3 +68,85 @@ def run_opt_neutral_set_dihedral(mol_name, mol_smile, functional='B3LYP', basis_
+
+    mol_dic : dictionary of oligomers where key is the name of the oligomer and value is the SMILES string
+
+    job_type : Type of job ran e.g. pop_opt_n
+
+    functional : Preset is B3LYP
+
+    basis_set1 : Preset 3-21G* (Lowest basis set)
+
+    basis_set2 : Preset 6-31G* (Median basis set)
+
+    basis_set3 : Preset 6-31G** (Highest basis set)
+
+    Returns three dictionaries:
+
+        fully_complete : Oligomers that have been calculated at the highest basis set
+
+        not_complete : Oligomers that failed and need manual assessment (shows basis set they failed at)
+
+        in_progress : Oligomers that are still in progress (shows basis set they are currently being ran at)
+
+
+    """
+    fully_complete = {}
+    not_complete = {}
+    in_progress = {}
+    k = mol_name
+
+    if os.path.isfile(f'{k}/{k}_{job_type}.log') == False:
+        #Then run it
+        run_opt_n(k, mol_dic, functional, basis_set1)
+        in_progress[f'{k}'] = basis_set1
+
+    if os.path.isfile(f'{k}/{k}_{job_type}.log') == True:
+    #If the file exists
+        data = cclib.ccopen(f'{k}/{k}_{job_type}.log').parse()
+        #parse that file
+        if data.metadata['success'] == False:
+            not_complete[f'{k}'] = basis_set1
+            #add to fail fictionary
+        if data.metadata['success'] == True:
+            #did the calculations work
+            if os.path.isfile(f'{k}/{k}_{job_type}.log') == False:
+                #Run the calc
+                run_opt_n(k, mol_dic, functional, basis_set2)
+                in_progress[f'{k}'] = basis_set2
+
+            if os.path.isfile(f'{k}/{k}_{job_type}.log') == True:
+                #Then check the next basis set worked or not
+                data2 = cclib.ccopen(f'{k}/{k}_{job_type}.log').parse()
+                if data2.metadata['success'] == False:
+                    #if the calc says failed then return not complete
+                    not_complete[f'{k}'] = basis_set2
+
+                if data2.metadata['success'] == True:
+                    #checking calculation worked
+                    if os.path.isfile(f'{k}/{k}_{job_type}.log') == False:
+                        #Run the calc
+                        run_opt_n(k, mol_dic, functional, basis_set3)
+                        in_progress[f'{k}'] = basis_set3
+
+                    if os.path.isfile(f'{k}/{k}_{job_type}.log') == True:
+                        #Checking if the final basis set worked or not
+                        data3 = cclib.ccopen(f'{k}/{k}_{job_type}.log').parse()
+                        if data3.metadata['success'] == True:
+                            fully_complete[f'{k}'] = basis_set3
+                            #All calculated and fine
+                        if data3.metadata['success'] == False:
+                            not_complete[f'{k}'] = basis_set3
+
+
+    return fully_complete, not_complete, in_progress
```

Comment:
Adds a function which is initial attempt to stage calculations. I think this could definitely be improved and streamlined but it works.